test/compress/issue-208.js

```diff
@@ -27,3 +27,28 @@ do_update_rhs: {
         MY_DEBUG += 0;
     }
 }
+
+mixed: {
+    options = {
+        global_defs: { DEBUG: 0 }
+    }
+    input: {
```

Comment:
> Having `const DEBUG = false;` at the top of the file and overriding it with `--define DEBUG=true` (or vice versa) would be a common use case.

Ah, I see. So we do want to actually override the definition? Currently (master & this PR) that `const` definition would simply make `global_defs` an no-op.